lib/board.rb

```diff
@@ -0,0 +1,80 @@
+require "pry"
+
+class Board
+
+  attr_accessor :cells, :player, :token_1, :token_2
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def reset! #Can reset the state of the board to how it should be at the start of the game an array with 9 " " elements.
+    @cells = Array.new(9, " ")
+  end
+
+  def display #Prints the current state of the board.
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(players_pick) #takes in the user's input in the form of 1-9 strings like "2" or "9" and looks up the value of the cells at the correct index from the array's perspective.
+    self.cells[players_pick.to_i-1]
+  end
+
+  def update(players_pick, player) #represents updating the board when a player makes a move
+
+    self.cells[players_pick.to_i-1] = player.token
+  end
+
+  def full? #True if board is entirely filled with "x/o"s
+    !@cells.include?(" ")
+
+  end
+
+  def turn_count
+    move_count = []
+    self.cells.each do |cell|
+      if cell != " "
+        move_count << cell
+      end
+    end
+    move_count.count
+  end
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell != " "}
end
```